src/steamship/client/steamship.py

```diff
@@ -144,7 +153,10 @@ def _instance_use(
             fetch_if_exists=fetch_if_exists,
         )
 
-        return instance
+    @staticmethod
+    def register_plugin_instance_subclass(plugin_handle: str, plugin_class: PluginInstance):
```

Comment:
If we're only planning to use this method during testing I would directly access the static variable in our test scripts instead of using a static function. 

In the tests this would look like: 

```
Steamship._PLUGIN_INSTANCE_SUBCLASS_OVERRIDES[plugin_handle] = plugin_class
```

Reply:
+1 to this- I think more thought would be necessary to figure out an extension pattern for user-submitted plugins.

Reply:
Will do!